browser/net/brave_request_handler.cc

```diff
@@ -99,6 +99,13 @@ void BraveRequestHandler::SetupCallbacks() {
       base::BindRepeating(brave::OnBeforeStartTransaction_SiteHacksWork);
   before_start_transaction_callbacks_.push_back(start_transaction_callback);
 
+#if BUILDFLAG(ENABLE_IPFS)
+  if (base::FeatureList::IsEnabled(ipfs::features::kIpfsFeature)) {
+    start_transaction_callback = base::BindRepeating(
```

Comment:
there's no reason to put this in a separate block, it can go here https://github.com/brave/brave-core/pull/14883/files#diff-ceffb3e0ca75b382f4f40500737a57be6618d89d201d6e6e39fc3231e10f5265R89

Reply:
also it's better to use a generic naming convention for IPFS rather than XForwardedProtoHeader because we might use this callback for other things in the future. `ipfs::OnBeforeStartTransaction` is sufficient

Reply:
sorry, actually this should be a URLLoaderThrottle as discussed in DM

Reply:
We've discussed in DM and it is ok to use brave_request_handler here